src/html/pretty.rs

```diff
@@ -0,0 +1,127 @@
+// Based on the implementation from `htmlq`: https://github.com/mgdm/htmlq/blob/6e31bc814332b2521f0316d0ed9bf0a1c521b6e6/src/pretty_print.rs
+// Available under the MIT License.
+// Copyright (c) 2019 Michael Maclean
+
+use std::{
+    collections::HashSet,
+    io::{self, Write},
+    str,
+};
+
+use html5ever::{
+    serialize::{HtmlSerializer, Serialize, SerializeOpts, Serializer, TraversalScope},
+    QualName,
+};
+
+use markup5ever::serialize::AttrRef;
+use once_cell::sync::Lazy;
+use scraper::Html;
+
+pub fn pretty_print(html: &Html) -> String {
+    let mut content: Vec<u8> = Vec::new();
+    let mut pp = PrettyPrint {
+        indent: 0,
+        previous_was_block: false,
+        inner: HtmlSerializer::new(
+            &mut content,
+            SerializeOpts {
+                traversal_scope: TraversalScope::IncludeNode,
+                ..Default::default()
+            },
+        ),
+        at_beginning: true,
+    };
+    Serialize::serialize(html, &mut pp, TraversalScope::IncludeNode).unwrap();
+    str::from_utf8(content.as_ref()).unwrap().to_owned()
+}
+
+static INLINE_ELEMENTS: Lazy<HashSet<&'static str>> = Lazy::new(|| {
```

Comment:
What's the behavior with these elements? Are they not stripped? Does it make sense to mention it?

Reply:
That is for the pretty printer, it is designed to work on any HTML.

The listed elements are printed on a single line instead of expanded, I'll add a comment.